clang/lib/Sema/SemaDecl.cpp

```diff
@@ -13456,6 +13458,15 @@ void Sema::AddInitializerToDecl(Decl *RealDecl, Expr *Init, bool DirectInit) {
     return;
   }
 
+  if (VDecl->isInvalidDecl()) {
+    CorrectDelayedTyposInExpr(Init, VDecl);
+    ExprResult Recovery =
+        CreateRecoveryExpr(Init->getBeginLoc(), Init->getEndLoc(), {Init});
+    if (Expr *E = Recovery.get())
```

Comment:
Oops. I think this is still UB, please double check:
```suggestion
    if (ExprResult Recovery =
         CreateRecoveryExpr(Init->getBeginLoc(), Init->getEndLoc(), {Init}))
         VDecl->setInit(Recovery.get());
```